### PR TITLE
Initialise quic-p2p

### LINKED ISSUE #711: Add Quic-P2P for client connections

### PR DESCRIPTION
**Pull request by octol**
Resolves #711

### CODE REVIEW

File: src/lib.rs [hunk 1]
```diff
@@ -57,6 +57,9 @@
 // TODO - Fix before committing.
 #![allow(unused)]
 
+#[macro_use]
+extern crate unwrap;
```
Comment (Fraser999) [inline src/lib.rs]: We can just do `use unwrap::unwrap;` now for importing macros.

File: src/source_elder.rs [hunk 2]
```diff
@@ -6,13 +6,44 @@
...
-impl SourceElder {}
+impl SourceElder {
+    pub fn new(config: QuickP2pConfig) -> (Self, Receiver<Event>) {
+        let (quic_p2p, event_receiver) = {
+            let (event_sender, event_receiver) = channel();
+            let mut quic_p2p = unwrap!(quic_p2p::Builder::new(event_sender)
+                .with_config(config)
+                .build());
+            let our_conn_info = unwrap!(quic_p2p.our_connection_info());
+            info!("QuickP2p started o {}", our_conn_info.peer_addr);
```
Comment (Fraser999) [inline src/source_elder.rs]: Should "o" be "on"?

File: src/source_elder.rs [hunk 3]
```diff
@@ -6,13 +6,44 @@
...
+            info!(
```
Comment (Fraser999) [inline src/source_elder.rs]: Not sure if we need both of these info messages?  Maybe this more verbose one should be `trace!`?  I'm not even sure what extra info is printed by outputting the whole `our_conn_info` vs just our addr from the previous log message.  If it's not useful, I'd probably just vote to remove this line.

File: src/vault.rs [hunk 4]
```diff
@@ -10,36 +10,54 @@ use crate::{
...
+            //id: Default::default(),
+            state: State::Elder { src },
+            event_receiver,
+        })
     }
 
     /// Run the main event loop.  Blocks until the vault is terminated.
     pub fn run(&mut self) {
-        unimplemented!();
+        info!("Running event loop");
```
Comment (Fraser999) [inline src/vault.rs]: Is this worthwhile keeping?  If you feel so, maybe downgrading to `trace!` would be more appropriate?

File: src/source_elder.rs (hunk 3 above)
Comment (octol) [reply src/source_elder.rs]: I think you need the output from `our_conn_info` for a client to connect. (It adds things like cert into)

What about upgrading the first one to `println`? Here is the output btw
```
/s/m/safe_vault (add-quic-p2p|✚2) $ env RUST_LOG=safe_vault=info cargo run
   Compiling safe_vault v0.18.0 (/home/jon/src/maidsafe/safe_vault)
    Finished dev [unoptimized + debuginfo] target(s) in 3.58s
     Running `target/debug/safe_vault`
[2019-06-13T15:41:17Z INFO  safe_vault] 

Running safe_vault v0.18.0
==========================
QuickP2p started on 127.0.0.1:55880
[2019-06-13T15:41:17Z INFO  safe_vault::source_elder] Our connection info:
{"peer_addr":"127.0.0.1:55880","peer_cert_der":[48,130,1,78,48,129,246,160,3,2,1,2,2,1,42,48,10,6,8,42,134,72,206,61,4,3,2,48,33,49,31,48,29,6,3,85,4,3,12,22,114,99,103,101,110,32,115,101,108,102,32,115,105,103,110,101,100,32,99,101,114,116,48,34,24,15,49,57,55,53,48,49,48,49,48,48,48,48,48,48,90,24,15,52,48,57,54,48,49,48,49,48,48,48,48,48,48,90,48,33,49,31,48,29,6,3,85,4,3,12,22,114,99,103,101,110,32,115,101,108,102,32,115,105,103,110,101,100,32,99,101,114,116,48,89,48,19,6,7,42,134,72,206,61,2,1,6,8,42,134,72,206,61,3,1,7,3,66,0,4,19,138,56,209,103,43,43,137,204,168,232,12,157,104,46,177,54,74,140,143,96,125,204,184,200,74,214,106,109,247,241,121,112,48,151,63,102,100,240,58,165,97,221,240,51,34,187,139,198,26,211,3,73,60,39,16,166,155,55,177,87,169,146,105,163,27,48,25,48,23,6,3,85,29,17,4,16,48,14,130,12,77,97,105,100,83,65,70,69,46,110,101,116,48,10,6,8,42,134,72,206,61,4,3,2,3,71,0,48,68,2,32,103,195,72,185,171,224,67,164,246,193,158,244,229,212,33,125,86,185,225,183,81,226,202,39,65,103,52,154,79,223,183,199,2,32,24,130,95,136,134,163,98,34,124,153,213,134,92,116,165,180,46,73,35,240,90,30,82,209,154,15,145,245,9,0,112,175]}

[2019-06-13T15:41:17Z INFO  safe_vault::vault] Running event loop
```

File: src/vault.rs (hunk 4 above)
Comment (octol) [reply src/vault.rs]: I'll remove it!

File: src/lib.rs (hunk 1 above)
Comment (ustulation) [reply src/lib.rs]: But if we need to do that in every module is it better to just stick to `#[macro_use]` ?

File: src/source_elder.rs (hunk 3 above)
Comment (ustulation) [reply src/source_elder.rs]: `println!` to me sounds better suited for this (as it's not really a loggin we are doing so much as compulsorily letting the user know what to give to others in order to connect to us). But IDK if there're any std practices/guidelines for these.

File: src/lib.rs (hunk 1 above)
Comment (Fraser999) [reply src/lib.rs]: That's not what we're doing in other places/crates.

Also, I guess if that's what we need to do to import functions and types, it makes sense for macros too.

File: src/source_elder.rs (hunk 3 above)
Comment (Fraser999) [reply src/source_elder.rs]: Yeah +1 for `println!` then from me too.

**Pull request merged**